tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/ToEtcExpr.scala

```diff
@@ -210,11 +220,25 @@ class ToEtcExpr(annotationStore: AnnotationStore, aliasSubstitution: ConstSubsti
         // x = y, x /= y
         val a = varPool.fresh
         val opsig = OperT1(Seq(a, a), BoolT1())
-        mkExRefApp(opsig, args)
+        val etcExpr = mkExRefApp(opsig, args)
+        etcExpr.typeErrorExplanation = (expectedTypes: List[TlaType1], actualTypes: List[TlaType1]) => {
+          s"Arguments of equality should have the same type\nFor arguments ${args.mkString(", ")} with types ${actualTypes
+              .mkString(", ")}\nIn expression $ex"
```

Comment:
```suggestion
          s"The arguments ${args.mkString(", ")} in $ex should have the same type. Expected: ${expectedTypes.mkString(", ")}, found ${actualTypes.mkString(", ")}."
```

Reply:
It was actually @konnov 's suggestion to explicitly say that equality requires arguments of same type and keep the expression at the end of the message in case it is too big: https://github.com/informalsystems/apalache/pull/1363#discussion_r809379594

Reply:
Yes, I know, but `\n` also breaks end-of-line tags (`E@16:42:30.894`). If you do want to keep the linebreaks, I'd at least change the capitalization, because it reads very awkwardly as is:
```suggestion
          s"Arguments of ${disPrefix}equality should have the same type.\nFound arguments ${args.mkString(", ")} with types ${actualTypes
              .mkString(", ")}\nin expression $ex."
```
where `disPrefix` is `"dis"` if the operator is `ne` and `""` otherwise.

Reply:
I've removed the linebreaks but kept the original structure like you did on this last suggestion